tests/OK/729.lp

```diff
@@ -0,0 +1,6 @@
+require tests.OK.bool as B;
+symbol T : (B.B → B.B) → TYPE;
+symbol bool_neg: B.B → B.B;
+symbol foo : T B.bool_neg;
+symbol bar : T bool_neg;
+assertnot ⊢ T B.bool_neg ≡ T bool_neg;
```

Comment:
Such tests should be performed on the output, with [CRAM tests](https://dune.readthedocs.io/en/stable/tests.html#cram-tests), but it requires some work.